tests/core/pyspec/eth2spec/test/altair/validator/test_validator.py

```diff
@@ -0,0 +1,155 @@
+    active_pubkeys = [state.validators[index].pubkey for index in validator_indices]
+
+    ensure_assignments_in_sync_committee(
+        spec, state, query_epoch, state.current_sync_committee, active_pubkeys
+    )
+    ensure_assignments_in_sync_committee(
+        spec, state, next_query_epoch, state.next_sync_committee, active_pubkeys
+    )
+
+    assert validator_count >= 3
+    sync_committee_pubkeys = set(
+        list(state.current_sync_committee.pubkeys)
+        + list(state.next_sync_committee.pubkeys)
+    )
+    disqualified_pubkeys = list(
+        filter(lambda key: key not in sync_committee_pubkeys, active_pubkeys)
+    )
+    some_pubkeys = rng.sample(disqualified_pubkeys, 3)
+    for pubkey in some_pubkeys:
+        validator_index = active_pubkeys.index(pubkey)
+        is_current = spec.is_assigned_to_sync_committee(
+            state, query_epoch, validator_index
+        )
+        is_next = spec.is_assigned_to_sync_committee(
+            state, next_query_epoch, validator_index
+        )
+        is_current_or_next = is_current or is_next
+        assert not is_current_or_next
+
+
+def _get_sync_committee_signature(
+    spec,
+    state,
+    target_slot,
+    target_block_root,
+    subcommittee_index,
+    index_in_subcommittee,
+):
+    subcommittee_size = spec.SYNC_COMMITTEE_SIZE // spec.SYNC_COMMITTEE_SUBNET_COUNT
+    sync_committee_index = (
+        subcommittee_index * subcommittee_size + index_in_subcommittee
+    )
+    pubkey = state.current_sync_committee.pubkeys[sync_committee_index]
+    privkey = pubkey_to_privkey[pubkey]
+
+    domain = spec.get_domain(
+        state,
+        spec.DOMAIN_SYNC_COMMITTEE,
+    )
+    signing_data = spec.compute_signing_root(target_block_root, domain)
+    return bls.Sign(privkey, spec.hash_tree_root(signing_data))
+
+
+@only_with_bls()
+@with_all_phases_except([PHASE0])
+@with_state
+def test_process_sync_committee_contributions(phases, spec, state):
+    # skip over slots at genesis
+    transition_to(spec, state, state.slot + 3)
+
+    # build a block and attempt to assemble a sync aggregate
+    # from some sync committee contributions
+    block = build_empty_block(spec, state)
+    previous_slot = state.slot - 1
+    target_block_root = spec.get_block_root_at_slot(state, previous_slot)
+    aggregation_bits = Bitvector[
+        spec.SYNC_COMMITTEE_SIZE // spec.SYNC_COMMITTEE_SUBNET_COUNT
+    ]()
+    aggregation_index = 0
+    aggregation_bits[aggregation_index] = True
+
+    contributions = [
+        spec.SyncCommitteeContribution(
+            slot=block.slot,
+            beacon_block_root=target_block_root,
+            subcommittee_index=i,
+            aggregation_bits=aggregation_bits,
+            signature=_get_sync_committee_signature(
+                spec, state, previous_slot, target_block_root, i, aggregation_index
+            ),
+        )
+        for i in range(spec.SYNC_COMMITTEE_SIZE // spec.SYNC_COMMITTEE_SUBNET_COUNT)
```

Comment:
I think it should be:

```suggestion
        for i in range(spec.SYNC_COMMITTEE_SUBNET_COUNT)
```

Reply:
ah yes, this was the intention, thanks for catching this!